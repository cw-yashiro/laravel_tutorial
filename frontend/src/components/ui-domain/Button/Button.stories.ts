import { Button } from "./Button"

export default {
	title: "ui-domain/Button",
	component: Button
}

export const Default = {
	args: {
		label: "ボタン",
		onClick: () => { }
	}
}