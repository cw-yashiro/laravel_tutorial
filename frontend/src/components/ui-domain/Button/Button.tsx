import { forwardRef } from "react"
import style from "./style.module.css"

type Props = {
  variant?: "primary" | "secondary" | "danger"
} & React.ComponentPropsWithRef<"button">

export const Button = forwardRef(({ variant = "primary", ...props }: Props) => {
  return <button className={style.module} {...props} data-variant={variant} />
})
