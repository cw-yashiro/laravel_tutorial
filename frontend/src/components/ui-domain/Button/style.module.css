.module {
  border: none;
  padding: 5px 10px;
  border-radius: 5px;
}

.module:hover {
  opacity: 0.9;
}

.module[data-variant="primary"] {
  background-color: #0d6efd;
  color: #fff;
}

.module[data-variant="secondary"] {
  background-color: #6c757d;
  color: #fff;
}

.module[data-variant="danger"] {
  background-color: #dc3545;
  color: #fff;
}
