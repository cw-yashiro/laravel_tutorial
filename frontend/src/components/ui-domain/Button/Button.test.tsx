import { describe, it, expect } from 'vitest'
import { screen, render } from '@testing-library/react'
import { Button } from './Button'

describe('ui-domain/Button', () => {
	it('buttonロールを持っていること', () => {
		render(<Button variant="primary" label="ボタン" onClick={() => { }} />)
		expect(screen.getByRole('button', { name: "ボタン" })).toBeInTheDocument()
	})
})