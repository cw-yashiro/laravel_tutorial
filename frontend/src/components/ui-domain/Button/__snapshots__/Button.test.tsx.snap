// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui-domain/Button > Snapshot: ボタンラベルが表示されている 1`] = `
<div>
  <button
    class="_module_3e7089"
    data-variant="primary"
  >
    ボタンラベル
  </button>
</div>
`;
